src/conn/conn_log.c

```diff
@@ -392,9 +392,11 @@ __log_archive_once(WT_SESSION_IMPL *session, uint32_t backup_file)
                                     WT_MIN(log->ckpt_lsn.l.file, backup_file);
 
     /* Adjust the number of log files to retain based on debugging options. */
-    if (conn->debug_ckpt_cnt != 0)
-        min_lognum = WT_MIN(conn->debug_ckpt[conn->debug_ckpt_cnt - 1].l.file, min_lognum);
-    if (conn->debug_log_cnt != 0) {
+    dbg_val = conn->debug_ckpt_cnt;
+    if (conn->debug_ckpt_enabled && dbg_val != 0)
+        min_lognum = WT_MIN(conn->debug_ckpt[dbg_val - 1].l.file, min_lognum);
+    dbg_val = conn->debug_log_cnt;
+    if (dbg_val != 0) {
```

Comment:
You might need `WT_ORDERED_READ` here?

Nothing prevents the compiler from replacing all of your uses of the local variable `dbg_val` with separate, from-main-memory reads of `conn.debug_log_cnt`.